packages/keystatic/src/form/fields/array/ui.tsx

```diff
@@ -4,78 +4,89 @@ import { ItemDropTarget, useDragAndDrop } from '@keystar/ui/drag-and-drop';
 import { FieldLabel, FieldMessage } from '@keystar/ui/field';
 import { Icon } from '@keystar/ui/icon';
 import { trash2Icon } from '@keystar/ui/icon/icons/trash2Icon';
-import { Flex } from '@keystar/ui/layout';
+import { VStack } from '@keystar/ui/layout';
 import { Item, ListView } from '@keystar/ui/list-view';
 import { Content } from '@keystar/ui/slots';
 import { Tooltip, TooltipTrigger } from '@keystar/ui/tooltip';
 import { Heading, Text } from '@keystar/ui/typography';
 import { useLocalizedStringFormatter } from '@react-aria/i18n';
-
-import { previewPropsToValue, valueToUpdater } from '../../get-value';
-import { createGetPreviewProps } from '../../preview-props';
+import { useField } from '@react-aria/label';
+import { useId, Key, useMemo, useState } from 'react';
 
 import l10nMessages from '../../../app/l10n/index.json';
-import { useId, Key, useMemo, useState } from 'react';
-import { getSlugFromState } from '../../../app/utils';
+import { getSlugFromState, pluralize } from '../../../app/utils';
+
+import { ArrayField, ComponentSchema, GenericPreviewProps } from '../../api';
 import { clientSideValidateProp } from '../../errors';
 import {
   FormValueContentFromPreviewProps,
   ExtraFieldInputProps,
 } from '../../form-from-preview';
+import { previewPropsToValue, valueToUpdater } from '../../get-value';
 import { getInitialPropsValue } from '../../initial-values';
+import { createGetPreviewProps } from '../../preview-props';
+
 import { useEventCallback } from '../document/DocumentEditor/ui-utils';
-import { ArrayField, ComponentSchema, GenericPreviewProps } from '../../api';
 
 export function ArrayFieldInput<Element extends ComponentSchema>(
   props: GenericPreviewProps<ArrayField<Element>, unknown> &
     ExtraFieldInputProps
 ) {
-  const labelId = useId();
-  const descriptionId = useId();
+  const minLength = props.schema.validation?.length?.min ?? 0;
+  const formId = useId();
   const stringFormatter = useLocalizedStringFormatter(l10nMessages);
+  const errorMessage = useArrayFieldValidationMessage(props);
+  const {
+    descriptionProps,
+    errorMessageProps,
+    fieldProps: groupProps,
+    labelProps,
+  } = useField({
```

Comment:
Prefer `useField()` here; let it handle the `aria-describedby` logic when combination of some, none or all description + validation message present.